Classes/New Issue/IssueTemplates.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  IssueTemplates.swift
+//  Freetime
+//
+//  Created by Ehud Adler on 11/3/18.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+import GitHubAPI
+import GitHubSession
+import Squawk
+
+struct IssueTemplate {
+    let title: String
+    let template: String
+}
+
+struct IssueTemplateDetails {
+    let repo: RepositoryDetails
+    let session: GitHubUserSession?
+    let viewController: UIViewController
+    weak var delegate: NewIssueTableViewControllerDelegate?
+}
+
+enum IssueTemplateHelper {
+
+    static private func matches(
+        regex: String,
+        text: String) -> [String] {
+        do {
+            let regex = try NSRegularExpression(pattern: regex)
+            let str = text as NSString
+            let results = regex.matches(
+                in: text,
+                range: NSMakeRange(0, str.length))
+            return results.map { str.substring(with: $0.range)}
+
+        } catch let error as NSError {
+            print("invalid regex: \(error.localizedDescription)")
+            return []
+        }
+    }
+
+    static func getNameAndDescription(fromTemplatefile file: String) -> (name: String?, about: String?) {
+        let names = IssueTemplateHelper.matches(regex: "(?<=name:).*", text: file)
+        let abouts = IssueTemplateHelper.matches(regex: "(?<=about:).*", text: file)
+        let name = names.first?.trimmingCharacters(in: .whitespaces)
+        let about = abouts.first?.trimmingCharacters(in: .whitespaces)
+        return (name, about)
+    }
+
+    static func showIssueAlert(
+        with templates: [IssueTemplate],
+        details: IssueTemplateDetails) -> UIAlertController {
+
+        let alertView = UIAlertController(
+            title: NSLocalizedString("New Issue", comment: ""),
+            message: NSLocalizedString("Choose Template", comment: ""),
+            preferredStyle: .alert
+        )
+
+        for template in templates {
```

Comment:
Correct me if I'm wrong, there appears to be no way to create an empty issue anymore? Is that correct?

As per the image below, GitHub has a button which allows you to just create an empty issue without a template!

![image](https://user-images.githubusercontent.com/15193942/48300953-ebd04380-e4dd-11e8-8db8-d509b636efba.png)